checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,406 @@
+import org.checkerframework.framework.type.treeannotator.ListTreeAnnotator;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+import org.checkerframework.framework.type.typeannotator.ListTypeAnnotator;
+import org.checkerframework.framework.type.typeannotator.TypeAnnotator;
+import org.checkerframework.javacutil.BugInCF;
+import org.checkerframework.javacutil.ElementUtils;
+import org.checkerframework.javacutil.TreeUtils;
+import org.checkerframework.javacutil.UserError;
+
+/** The annotated type factory for the Called Methods checker. */
+public class CalledMethodsAnnotatedTypeFactory extends AccumulationAnnotatedTypeFactory {
+
+    /**
+     * The {@link java.util.Collections#singletonList} method. It is treated specially by {@link
+     * #adjustMethodNameUsingValueChecker(String, MethodInvocationTree)}.
+     */
+    private final ExecutableElement collectionsSingletonList;
+
+    /**
+     * Whether to use the Value Checker as a subchecker to reduce false positives when analyzing
+     * calls to the AWS SDK. Defaults to false. Controlled by the command-line option {@code
+     * -AuseValueChecker}.
+     */
+    private final boolean useValueChecker;
+
+    /**
+     * The builder frameworks (such as Lombok and AutoValue) supported by the Called Methods
+     * checker.
+     */
+    private Collection<BuilderFrameworkSupport> builderFrameworkSupports;
+
+    /**
+     * Lombok has a flag to generate @CalledMethods annotations, but they used the old package name,
+     * so we maintain it as an alias.
+     */
+    private static final String OLD_CALLED_METHODS =
+            "org.checkerframework.checker.builder.qual.CalledMethods";
+
+    /**
+     * Lombok also generates an @NotCalledMethods annotation, which we have no support for. We
+     * therefore treat it as top.
+     */
+    private static final String OLD_NOT_CALLED_METHODS =
+            "org.checkerframework.checker.builder.qual.NotCalledMethods";
+
+    /**
+     * Create a new CalledMethodsAnnotatedTypeFactory.
+     *
+     * @param checker the checker
+     */
+    public CalledMethodsAnnotatedTypeFactory(BaseTypeChecker checker) {
+        super(
+                checker,
+                CalledMethods.class,
+                CalledMethodsBottom.class,
+                CalledMethodsPredicate.class);
+        Set<String> disabledFrameworks = new HashSet<>();
+        if (checker.hasOption(CalledMethodsChecker.DISABLE_BUILDER_FRAMEWORK_SUPPORTS)) {
+            disabledFrameworks.addAll(
+                    Arrays.asList(
+                                    checker.getOption(
+                                                    CalledMethodsChecker
+                                                            .DISABLE_BUILDER_FRAMEWORK_SUPPORTS)
+                                            .split(","))
+                            .stream()
+                            .map(String::toUpperCase)
```

Comment:
Don't normalize case.  It is confusing that the examples in the documentation and in the tests are different.  Permitting arbitrary casing leads to such confusion, and makes it more difficult for a user to find all uses of a particular string in their code or the code of other people (because it could appear in multiple different forms).  It does not prevent any hidden problems, because this code warns if a user misspells a framework name.  In other words, I see disadvantages but no benefits to this undocumented feature.